packages/app/src/specs/InlineSpecList.vue

```diff
@@ -87,4 +113,49 @@ const specs = computed(() => {
   )
 })
 
+const renderTeleport = ref(false)
+const showList = ref(true)
+let teleportMutuationObserver: MutationObserver | null = null
+
+onKeyStroke('f', () => {
+  showList.value = !showList.value
+})
+
```

Comment:
This seems like a very convoluted way of doing this. Why are we teleporting this button from Vue to React at all? I don't think this makes a ton of sense - all the checks, mutation observer and usage of `nextTick` are a bit of code smell to me.

I also don't think we should mix React and Vue like this where possible.

We already have a pattern for communicating between different parts of the runner UI (runner, reporter, driver), which is the `eventManager`.

What I'd suggest:

- In reporter, create the button using React. When you click it, you will do `events.emit('hide:specs')`, like [here](https://github.com/cypress-io/cypress/blob/10.0-release/packages/reporter/src/header/header.tsx#L24) for example.

- In the reporter events, https://github.com/cypress-io/cypress/blob/10.0-release/packages/reporter/src/lib/events.ts#L241, add something like

```ts
 localBus.on('hide:specs', (cb) => {
      runner.emit('hide:specs', cb)
    })
```

Confusing, `runner` refers to`eventManager` here, I'm fairly sure.

Now listen for that event in `eventManger`, here: https://github.com/cypress-io/cypress/blob/10.0-release/packages/app/src/runner/event-manager.ts#L200 create something like

```ts
reporterBus.on('hide:specs', () => {
  this.emit('hide:specs')
})
```

Finally, then listen for that event in `SpecRunner.vue`, like this: https://github.com/cypress-io/cypress/blob/10.0-release/packages/app/src/runner/SpecRunner.vue#L116-L118

```ts
const runnerStore = useRunnerStore()

eventManager.on('hide:specs', () => {
  runnerStore.toggleSpecList()
})
```

Note that `runnerStore` doesn't exist, but I think we want to create a new store for all the runner UI state - there's a lot more coming. The stores are here: https://github.com/cypress-io/cypress/tree/10.0-release/packages/app/src/store